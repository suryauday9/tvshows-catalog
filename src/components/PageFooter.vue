<template>
  <v-footer color="blue-grey darken-3">
    <div id="nav" class="nav-links">
      <router-link to="/">Home</router-link>
    </div>
    <v-col class="text-right white--text">
      <strong>copyright &copy; 2021 </strong>
    </v-col>
  </v-footer>
</template>

<script>
export default {
  name: "PageFooter"
};
</script>
<style scoped>
#nav a {
  color: #1976d2;
}
</style>
