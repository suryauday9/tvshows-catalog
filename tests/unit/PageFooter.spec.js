// Components
import PageFooter from '@/components/PageFooter.vue';

// Utilities
import { initializeApp } from './initializeApp';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = initializeApp(createLocalVue())

describe('PageFooter view', () => {
  let wrapper;
  const mountComponent = (options) => {
    return shallowMount(PageFooter, {
        localVue,
        ...options,
    })
  };

  beforeEach(() => {
      wrapper = mountComponent();
  })

  afterEach(() => {
      wrapper.destroy();
  })

  it('renders component correctly', () => {
      // see if component renders
      expect(wrapper.element).toMatchSnapshot();
  });

  it('should be a Vue instance', () => {
      expect(wrapper.isVueInstance).toBeTruthy();
  });

})
