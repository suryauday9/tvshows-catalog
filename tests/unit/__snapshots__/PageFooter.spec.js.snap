// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooter view renders component correctly 1`] = `
<v-footer-stub
  color="blue-grey darken-3"
  height="auto"
  tag="footer"
>
  <div
    class="nav-links"
    id="nav"
  >
    <router-link-stub
      ariacurrentvalue="page"
      event="click"
      tag="a"
      to="/"
    >
      Home
    </router-link-stub>
  </div>
   
  <v-col-stub
    class="text-right white--text"
    tag="div"
  >
    <strong>
      copyright Â© 2021 
    </strong>
  </v-col-stub>
</v-footer-stub>
`;
